frontend/src/views/players/players.test.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import {render, unmountComponentAtNode} from "react-dom";
+import {act} from "react-dom/test-utils";
+import {BrowserRouter as Router} from 'react-router-dom';
+import PlayersPage from "./players";
+import {playersApi} from "./players_api";
+
+jest.mock('./players_api', () => ({
+  playersApi: {
+    getAll: jest.fn(() => [
+      {code:'p1', description: "John", email: 'john@example.com'},
+      {code:'p2', description: "Doe", email: 'doe@example.com'}
+    ])
+  }
+}));
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useRouteMatch: () => ({url: '/players'}),
+}));
+
+
+describe("Testing players API", () => {
+  it('Get all players', async () => {
+    // FIXME: this is not really testing any thing, as we've mocked the api anyway.
```

Comment:
I think we should have tests more close to reality as possible. With mocked data not only there's a maintenance cost (every time there's a model change we will have to recreate it) but it could potentially increase the tests permissiveness (since they are being executed with not real data). In other words I think is safer to test with real data 👍

Apart of component testing, shouldn't the tests include the api calls in its scope?

Reply:
Moved here: #25